_sources/Iteration/Exercises.rst

```diff
@@ -14,31 +14,139 @@
 
+        Of course, that's not quite right because Ouack and Quack are misspelled.
+        Can you fix it?
+        ~~~~
 
-Of course, that's not quite right because Ouack and Quack are misspelled.
-Can you fix it?
+        prefixes = "JKLMNOPQ"
+        suffix = "ack"
 
-.. actex:: ex6_11_1
+        for p in prefixes:
+            print(p + suffix)
 
 
-2. Get the user to enter some text and print it out in reverse order.
+.. question:: q6_11_2
 
-.. actex:: ex6_11_2
+    .. actex:: ex6_11_2
+
+        Get the user to enter some text and print it out in reverse order.
+        ~~~~
+
+.. question:: q6_11_3
+
+
+    .. actex:: ex6_11_3
+
+        Get the user to enter some text and print out True if it's a palindrome, False otherwise. (Hint: reuse
+        some of your code from the last question. The == operator compares two values to see if they are the same)
+        ~~~~
+
+.. question:: iter_ex_3
+
+    .. tabbed:: q3
+
+        .. tab:: Question
+
+
+           .. actex:: ex_3_3
+
+              Write a program that uses a for loop to print
+                 |  ``One of the months of the year is January``
+                 |  ``One of the months of the year is February``
+                 |  ``One of the months of the year is March``
+                 |  etc ...
+              ~~~~
+
+        .. tab:: Answer
+
+            .. activecode:: q3_answer
+
+
+                for amonth in ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'November', 'December']:
+                    print("One of the months of the year is", amonth)
+
+        .. tab:: Discussion
+
+            .. disqus::
+                :shortname: interactivepython
+                :identifier: b271442ee0864973a023c19f27aeb401
+
+
+.. question:: iter_ex_4
+
+
+   .. actex:: ex_3_4
+
+      Assume you have a list of numbers ``12, 10, 32, 3, 66, 17, 42, 99, 20``
+
+      a. Write a loop that prints each of the numbers on a new line.
+      b. Write a loop that prints each number and its square on a new line.
+      ~~~~
+
+.. question:: iter_ex_6
+
+   .. actex:: ex_3_6
+      :nocodelens:
+
+      Write a program that asks the user for the number of sides, the length of the side, the color, and the fill color of a
+      regular polygon.  The program should draw the polygon and then fill it in.
+      ~~~~
+
+
+.. question:: iter_ex_7
+
+   .. tabbed:: q7
+
+       .. tab:: Question
+
+            .. actex:: ex_3_7
+               :nocodelens:
+
+               A drunk pirate makes a random turn and then takes 100 steps forward, makes another random turn, takes another 100 steps, turns another random amount, etc.  A social science student records the angle of each turn before the next 100 steps are taken.  Her experimental data is ``160, -43, 270, -97, -43, 200, -940, 17, -86``. (Positive angles are counter-clockwise.)  Use a turtle to draw the path taken by our drunk friend.  After the pirate is done walking, print the current heading.
```

Comment:
Add hints? Try to make it work for any sequence of experimental data. Iteration and the accumulator pattern might help you.